reference/security.rst

```diff
@@ -0,0 +1,291 @@
+JAAS Security
+=============
+
+This document is an overview of JAAS specifically covering areas pertaining to security and/or sensitive data.
+
+In each section we explain how the information is stored/transmitted and any cryptographic
+technologies used.
+
+Cloud-Credentials
+-----------------
+
+Cloud-credentials are the API keys Juju/JAAS needs in order to provision cloud infrastructure.
+These keys are uploaded to and stored by JAAS. Whenever a a model is created, the key is uploaded to 
+the Juju controller and used to manage cloud resources.
+
+User provided cloud credentials are stored securely and safely within a secret `vault <https://www.vaultproject.io/>`__. 
+It is  essential to handle these secrets securely to prevent unauthorised access and data breaches
+to the user's cloud.
+
+The use of Vault ensures that credentials are encrypted at rest and provides tools to avoid unauthorised
+access.
+
+JAAS - Juju Communication
+--------------------------
+
+JAAS acts as a central auth gateway between users and Juju controllers.
+Juju controllers explicitly trust JAAS by setting the ``login-token-refresh-url`` at bootstrap
+time. See our :doc:`how-to <../how-to/add_controller>` for more info on how to setup a new
+Juju controller for JAAS.
+
+Trust between Juju controllers and JAAS is established through the use of asymmetric cryptography
+and `JSON Web Tokens <https://jwt.io/introduction>`__ (JWT). 
+
+.. hint::
+
+    To avoid confusion, note that JWTs are also used elsewhere in JAAS particularly in user sessions. 
+    Although solving a similar problem, the tokens used between JAAS and Juju and those issued to users
+    are two separate systems.
+
+The ``login-token-refresh-url`` config option points to a
+`JSON Web Key Set <https://auth0.com/docs/secure/tokens/json-web-tokens/json-web-key-sets>`__ 
+(JWKS) endpoint. This endpoint contains a set of public keys used to verify the authenticy of tokens
+issued by JAAS.
+
+Whenever a request is made by JAAS to a Juju controller, a JWT is issued that encodes the user's
+information and permissions. This mechanism allows the authorisation and authentication to be securely
+delegated from the Juju controller to JAAS.
+
+Specific details are below:
+
+- **JWKS endpoint:** ``<jimm-url>/.well-known/jwks.json``
+- **Key Type/Size:** RSA 4096 bits
+- **Signing Algorithm:** RS256
```

Comment:
For consistency:

s/Type/type
s/Size/size
s/Algorithm/algorithm

Reply:
Opted to keep them as all caps